crypto/crypto/src/ed25519.rs

```diff
@@ -108,8 +108,24 @@ impl Ed25519Signature {
         }
     }
 
-    /// Check for correct size and malleability issues.
-    /// This method ensures s is of canonical form and R does not lie on a small group.
+    /// Check for correct size and third-party based signature malleability issues.
+    /// This method is required to ensure that given a valid signature for some message under some
+    /// key, an attacker cannot produce another valid signature for the same message and key.
+    ///
+    /// According to [RFC8032](https://tools.ietf.org/html/rfc8032), signatures are of the form
+    /// (S, R) and we should enforce that S is of canonical form (smaller than L, where L is the
+    /// order of edwards25519 curve group) to prevent signature malleability. Without this check,
+    /// one could add a multiple of L into S and still pass signature verification, resulting in
+    /// malleable signatures.
+    ///
+    /// Regarding the R part of the signature, this method does not check if it lies in a
+    /// small group because R is hashed during signing and verification to compute
+    /// h = H(ENC(R) || ENC(A) || M), which means that a third-party cannot modify R without being
+    /// detected.
+    ///
+    /// Note: It's true that malicious signers can already produce varying signatures by
```

Comment:
I think the non-signer attack is the important issue in malleability, and the right mental model to have. /cc @mimoo to check if I missed something.

Reply:
agree, signatures are not unique in ed25519 so we can't really do much about a signer that wants to shoot themselves in the foot

Reply:
Note that in theory, there are business scenarios where malicious signers can potentially use malleability to their advantage, so it's not always the case that they shoot themselves in the foot. I.e., consider the following attack vector:

Signer picks a low order R on purpose, which will allow her to **NOT commit to a single message**. A possible incentive is to create confusion, did you send 10 Libra or 1000 Libra? (both transactions can be verified with the same signature).

However, this attack requires a low order PK as well. But, we already check that on every key deserialization.

Reply:
O_o how do you represent sending 10 or 100 libra with the same message?